NuKeeper/Commands/RepositoryCommand.cs

```diff
@@ -1,45 +1,62 @@
-using System;
 using McMaster.Extensions.CommandLineUtils;
 using NuKeeper.Abstractions.CollaborationPlatform;
 using NuKeeper.Abstractions.Configuration;
 using NuKeeper.Engine;
 using NuKeeper.Inspection.Logging;
+using System;
+using System.Collections.Generic;
 
 namespace NuKeeper.Commands
 {
     [Command(Description = "Performs version checks and generates pull requests for a single repository.")]
     internal class RepositoryCommand : GitHubNuKeeperCommand
     {
-
         [Argument(0, Name = "Repository URI", Description = "The URI of the repository to scan.")]
         public string GitHubRepositoryUri { get; set; }
 
-        public RepositoryCommand(ICollaborationEngine engine, IConfigureLogger logger, IFileSettingsCache fileSettingsCache, ISettingsReader settingsReader)
-            : base(engine, logger, fileSettingsCache, settingsReader)
+        private readonly IEnumerable<ISettingsReader> _settingsReaders;
+
+        public RepositoryCommand(ICollaborationEngine engine, IConfigureLogger logger, IFileSettingsCache fileSettingsCache, ICollaborationFactory collaborationFactory, IEnumerable<ISettingsReader> settingsReaders)
+            : base(engine, logger, fileSettingsCache, collaborationFactory)
         {
+            _settingsReaders = settingsReaders;
         }
 
         protected override ValidationResult PopulateSettings(SettingsContainer settings)
         {
-            var baseResult = base.PopulateSettings(settings);
-            if (!baseResult.IsSuccess)
+            if (!Uri.TryCreate(GitHubRepositoryUri, UriKind.Absolute, out var repoUri))
             {
-                return baseResult;
+                return ValidationResult.Failure($"Bad repository URI: '{GitHubRepositoryUri}'");
             }
 
-            if (!Uri.TryCreate(GitHubRepositoryUri, UriKind.Absolute, out var repoUri))
+            var didRead = false;
+            foreach (var reader in _settingsReaders)
             {
-                return ValidationResult.Failure($"Bad repository URI: '{GitHubRepositoryUri}'");
+                if (reader.CanRead(repoUri))
+                {
+                    didRead = true;
+                    settings.SourceControlServerSettings.Repository = reader.RepositorySettings(repoUri);
+                }
+            }
+
+            if (!didRead)
+            {
+                return ValidationResult.Failure($"Unable to work out which platform to use");
```

Comment:
Same here, the message sound mention the `GitHubRepositoryUri` that could not be matched